### PR TITLE
build-sys: add Makefile compatible with darvaza's but mono-project

### PR DESCRIPTION
**Pull request by amery**
same as darvaza's but without the subproject handling

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+GOBIN ?= $(GOPATH)/bin
+
+REVIVE ?= $(GOBIN)/revive
+REVIVE_FLAGS ?= -formatter friendly
+REVIVE_INSTALL_URL ?= github.com/mgechev/revive
+
+TMPDIR ?= .tmp
+
+all: get generate fmt tidy build
```
Comment (karasz) [inline Makefile:17]: Just wondering if fmt should not be after tidy. On the other hand tidy is not touching .go files only .mod and .sum as opposed to fmt which just touches .go files. So probably it is fine.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	$(GO) test -v ./...
+
+up:
+	$(GO) get -u -v ./...
+	$(GO) mod tidy
+	$(GO) install -v $(REVIVE_INSTALL_URL)
+
+generate:
+	@git grep -l '^//go:generate' | sed -n -e 's|\(.*\)/[^/]\+\.go$$|\1|p' | sort -u | while read d; do \
+		git grep -l '^//go:generate' "$$d"/*.go | xargs -r $(GO) generate $(GOGENERATE_FLAGS); \
```
Comment (karasz) [inline Makefile:46]: Why not go generate ./...?

**Review by karasz — APPROVED** [review]
LGTM comments are mostly cosmetic.

File: Makefile (hunk 1 above)
Comment (amery) [reply Makefile]: `tidy` already calls `fmt` so I've removed the explicit reference in `all`

File: Makefile (hunk 2 above)
Comment (amery) [reply Makefile]: I honestly don't remember and can't justify it. I've been carrying it around from other projects without asking myself if it was relevant. I've replaced it with a single go generate call

**Comment by amery** [comment]
@karasz amended

**Review by karasz — APPROVED** [review]
now looks perfect

**Pull request merged**